src/java/org/apache/cassandra/service/ActiveRepairService.java

```diff
@@ -828,6 +868,13 @@ public void handleMessage(Message<? extends RepairMessage> message)
 
         if (session == null)
         {
+            switch (message.verb())
+            {
+                case VALIDATION_RSP:
+                case SYNC_RSP:
+                    ctx.messaging().send(message.emptyResponse(), message.from());
```

Comment:
Was there an issue here where senders of `VALIDATION_RSP` and `SYNC_RSP` would sit around waiting for an ack that never came when the session couldn't be found at the target node?

Reply:
Also, do we need to send this response if the initiator is local?

Reply:
> Was there an issue here where senders of VALIDATION_RSP and SYNC_RSP would sit around waiting for an ack that never came when the session couldn't be found at the target node?

yes, tests found this bug.

> Also, do we need to send this response if the initiator is local?

No, but don't think we know that, so figure a ignored ack is fine